Automattic/woocommerce-payments
# feat: remove non-UPE methods when UPE is disabled

Fixes #

#### Changes proposed in this Pull Request









When UPE is disabled, the backend removes non-UPE payment methods from the gateway: https://github.com/Automattic/woocommerce-payments/blob/5aefa3eb392b990a5fde0d76e3c40aa323f8808b/includes/admin/class-wc-rest-upe-flag-toggle-controller.php#L126-L131
Which leaves the frontend with an inconsistent state, until the page is refreshed.

Duplicating the same logic to the backend on the frontend, to ensure non-UPE payment methods are removed when the flag is disabled.

#### Testing instructions





- Ensure you have "Enable UPE settings changes" enabled
- Ensure you have "Enable UPE checkout" enabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments
- Add a non-UPE payment method
- In the console, you should see it when querying `wp.data.select('wc/payments').getEnabledPaymentMethodIds()`
- Disable UPE
- The non-UPE payment methods is removed from the store
- You can check this by looking at the browser's console: `wp.data.select('wc/payments').getEnabledPaymentMethodIds()`

-------------------
